### PR TITLE
Add/lane splitting module fw 46

### PR DESCRIPTION
**Pull request by markstuart**
Ported in existing Lane splitter from prototype, modified to suit requirements

### CODE REVIEW

**Comment by markstuart** [comment]
Further PR's in the works for:
- Using some npm modules instead of handrolled stuff in `geoTools` lib
- Providing more configurability

File: lib/splitter.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  assert(isRoad(road), '`road` supplied must be a LineString with property highway one of `residential` or `primary`')
+
+  options = merge(DEFAULTS, options || {})
+
+  var featureCollection = {
+    type: 'FeatureCollection',
+    features: []
+  }
+
+  sanitizeRoadData(road)
```
Comment (matt-in-a-hat) [inline lib/splitter.js:132]: Wanna give this road a clone before butchering it?

**Comment by matt-in-a-hat** [comment]
Looking good

**Comment by digitalsadhu** [comment]
Nice work, I wish I had the slightest idea what this black magic voodoo was ;)

**Pull request merged**